src/actors/process_list/process_list.py

```diff
@@ -0,0 +1,76 @@
+import os
+import sys
+import json
+
+
+def readlink(path, filename):
+    return os.readlink(os.path.join(path, filename))
+
+
+def get_environ(pid):
+    try:
+        with open(os.path.join(pid, 'environ'), 'rb') as f:
+            output = {}
+            environ = f.read().split("\x00")
+            for env in environ:
+                if env:
+                    name, val = env.split('=', 1)
+                    output[name] = val
+            return output
+    except EnvironmentError as exc:
+        sys.stderr.write(str(exc) + "\n")
+        return None
+
+
+def get_cmdline(pid):
+    try:
+        with open(os.path.join(pid, 'cmdline'), 'rb') as f:
+            return f.read().replace("\x00", " ").strip()
+    except EnvironmentError as exc:
+        sys.stderr.write(str(exc) + "\n")
+        return None
+
+
+def extract_filter_param(params):
+    if params:
+        try:
+            return str(params['process_list_filter'][0]['value'])
+        except LookupError:
+            sys.stderr.write("Invalid filter param structure, continue without filtering...\n")
+    return ''
+
+
+if __name__ == '__main__':
+    params = {} if sys.stdin.isatty() else json.load(sys.stdin)
+    filter_param = extract_filter_param(params)
+
+    actor_output = {'process_list': [
+        {'processes': []}
+    ]}
+
+    root = '/proc'
+    cpid = str(os.getpid())
```

Comment:
I would like to get ```os.getpid``` only once.
Hmm I would like to keep root as a local variable, i am not sure if in that case this would be more idiomatic to have global constant.